tests/test_rag_edge_cases.py

```
@@ -0,0 +1,865 @@
+            urls=["https://example.com/new-page"],
+            pages=pages,
+        )
+
+        # Mock existing chunks to be deleted (from old content that no longer exists)
+        mock_vector_service.get_chunks_by_source.return_value = [
+            {"id": "orphan1", "content": "orphaned content 1", "content_hash": "hash1"},
+            {"id": "orphan2", "content": "orphaned content 2", "content_hash": "hash2"},
+        ]
+
+        # Process crawl result (should detect and delete orphans)
+        result = await service.process_crawl_result(
```

Ruff (0.12.2)

Local variable `result` is assigned to but never used

Remove assignment to unused variable `result`

(F841)